### PR TITLE
bearer token create and verify

### PR DESCRIPTION
**Pull request by phoebe-lew**

### CODE REVIEW

File: js/src/crypto.ts [hunk 1]
```diff
@@ -156,6 +179,132 @@ export class Crypto {
...
   }
+
+  /**
+   * signs the message as a compact jws with nonce
+   * @param privateKeyJwk - the key to sign with
+   * @param kid - the kid to include in the jws header. used by the verifier to select the appropriate verificationMethod
+   *              when dereferencing the signer's DID
+   * @returns the compact JWS
+   */
+  static async token(opts: TokenOptions): Promise<string> {
```
Comment (jiyoontbd) [inline js/src/crypto.ts:190]: what is the difference between this method and the `sign` method in line 85?

Comment (phoebe-lew) [reply js/src/crypto.ts:190]: This is for creating the bearer token for get requests and sign populates the `signature` field on a tbdex message

Comment (jiyoontbd) [reply js/src/crypto.ts:190]: ah, i see they're slightly different in implementation - sign() uses `Crypto.hash(jwspayload)` while token() does not. cool!

Comment (jiyoontbd) [reply js/src/crypto.ts:190]: i was originally thinking that they should be squished into one method, but that part is distinct

File: js/src/crypto.ts [hunk 2]
```diff
@@ -10,14 +11,15 @@ import * as cbor from 'cbor-x'
 import { sha256 } from '@noble/hashes/sha256'
 import { Convert } from '@web5/common'
 import { EcdsaAlgorithm, EdDsaAlgorithm, Jose } from '@web5/crypto'
-import { deferenceDidUrl, isVerificationMethod } from './did-resolver.js'
+import { DidResource, deferenceDidUrl, isVerificationMethod } from './did-resolver.js'
+import { DidDocument, VerificationMethod } from '@web5/dids'
```
Comment (mistermoe) [inline js/src/crypto.ts:15]:
```suggestion
```

File: js/src/crypto.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import type { MessageModel, ResourceModel } from './types.js'
+import type { BearerTokenModel } from './pfi-rest-client.js'
```
Comment (mistermoe) [inline js/src/crypto.ts:2]:
```suggestion
```

File: js/src/crypto.ts (hunk 1 above)
Comment (phoebe-lew) [reply js/src/crypto.ts:190]: eventually i wanna refactor to reduce duplication

**Review by jiyoontbd — APPROVED** [review]

**Comment by mistermoe** [comment]
Closing this out bc we pulled this code into #147

**Pull request closed without merging**